### DISCUSSION TITLE
Labeled R groups matching implicit H

### DISCUSSION THREAD

**Discussion by kulmes** [rdkit/rdkit#6126] [Q&A]
I'm trying to match a scaffold with implicit H's sometimes at the r group positions I specify, and not having much luck.  This is what I'm attempting to do:
`scaffold = Chem.MolFromSmiles('CC(N=C([*:4])N([*:3])[*:2])[*:1]')`
![image](https://user-images.githubusercontent.com/123330789/221044286-7a0c4149-c551-4829-a5cc-1ee335a7550e.png)
```
test = Chem.MolFromSmiles('CC(N=C(C)(NCC))C')
params = rdRGroupDecomposition.RGroupDecompositionParameters()
params.onlyMatchAtRGroups = True
params.removeAllHydrogenRGroups = False
params.removeHydrogensPostMatch = False
params.removeAllHydrogenRGroupsAndLabels = False
groups,_ = rdRGroupDecomposition.RGroupDecompose([scaffold],[test],asSmiles=False,asRows=False)
PandasTools.RGroupDecompositionToFrame(groups,test,include_core=True)
````
![image](https://user-images.githubusercontent.com/123330789/221043993-8a5452e8-7e58-4cfb-8ac6-74e762f95eb9.png)
As you can see I'm not getting a match for R3 = H.  I thought it had something to do with the removing of Hydrogens, or perhaps the fact that the hydrogens were implicit in the SMILES string I used to create the molecule, but even if I specify 'CC(N=C(C)N(CC)[H])C' as the test molecule I don't get an R3 match.  
Anyone have any ideas?

**Comment by greglandrum** [comment]
@kulmes you have forgotten to pass the parameters object to the call to `RGroupDecompose()`.
If you do
```
groups,_ = rdRGroupDecomposition.RGroupDecompose([scaffold],[test],asSmiles=False,asRows=False,options=params)
```
I believe you will get the answer you're looking for.
